Color output in console can conflict with output_runner regex

It would probably be best to have output_runner strip any color codes from the output before trying to match the regex.

Yes, I believe removing those escape codes would be the most general solution.

I am seeing output from rules_typescript that looks like:
`[96mpath/to/my/file.ts[0m:[93m1[0m:[93m48[0m - [91merror[0m[90m TS2307: [0mCannot find module 'library/file'.`
Which I am currently matching using the regex `(\\x1B\\[96m)?(.*?\\.ts).*?(\\x1B\\0m)?:(\\x1B\\[93m)?\\d+(\\x1B\\[0m)?:(\\x1B\\[93m)?\\d+(\\x1B\\[0m)? - (\\x1B\\[91m)?error(\\x1B\\[0m)?(\\x1B\\[90m)? TS2307: (\\x1B\\[0m)?Cannot find module '(.*?)'\\.`

For my use cases the regex simply replacing `/\x1B\[\d*?m/g` with nothing would work for the colors, I'm not sure but I think `/\x1B\[[0-9;]*[mGKHF]/g` would match all of the escape codes.

I also added a `.*?` because I was sometimes seeing a backspace character being output as well.  Would it make sense to also remove those?

--- a/ibazel/output_runner/output_runner.go
+++ b/ibazel/output_runner/output_runner.go
@@ -41,6 +41,9 @@ var runOutputInteractive = flag.Bool(
 	true,
 	"Use an interactive prompt when executing commands in Bazel output")
 
+// This RegExp will match ANSI escape codes.
+var escapeCodeCleanerRegex = regexp.MustCompile("\\x1B\\[[\\x30-\\x3F]*[\\x20-\\x2F]*[\\x40-\\x7E]")
+
 type OutputRunner struct{}
 
 type Optcmd struct {
@@ -114,7 +117,7 @@ func matchRegex(optcmd []Optcmd, output *bytes.Buffer) ([]string, []string, [][]
 	var args [][]string
 	scanner := bufio.NewScanner(output)
 	for scanner.Scan() {
-		line := scanner.Text()
+		line := escapeCodeCleanerRegex.ReplaceAllLiteralString(scanner.Text(), "")
 		for _, oc := range optcmd {
 			re := regexp.MustCompile(oc.Regex)
 			matches := re.FindStringSubmatch(line)

--- a/ibazel/output_runner/output_runner_test.go
+++ b/ibazel/output_runner/output_runner_test.go
@@ -115,3 +115,48 @@ func TestMatchRegex(t *testing.T) {
 		}
 	}
 }
+
+var cleanerTests = []struct {
+	in  string
+	out []string
+}{
+	{
+		"buildozer 'add deps //wow' //fake:target0",
+		[]string{"buildozer 'add deps //wow' //fake:target0"},
+	},
+	{
+		"[96mbuildozer [0m[93m[0m[93m[0m[91m'add deps [0m[90m//wow'[0m //fake:target1",
+		[]string{"buildozer 'add deps //wow' //fake:target1"},
+	},
+	{
+		"build[96mozer 'add d[96meps //w[96mow' //fake:tar[96mget2",
+		[]string{"buildozer 'add deps //wow' //fake:target2"},
+	},
+	{
+		"[0m[90mbuildozer 'a[0m[93mdd deps //wow' [0m[93m//fake:target3[91m",
+		[]string{"buildozer 'add deps //wow' //fake:target3"},
+	},
+	{
+		"buildozer 'add deps //wow[0m[93m //fake:target4",
+		[]string(nil),
+	},
+}
+
+func TestMatchCleanRegex(t *testing.T) {
+	optcmd := []Optcmd{
+		{Regex: "^(buildozer) '(.*)'\\s+(.*)$", Command: "$1", Args: []string{"$2", "$3"}},
+	}
+
+	for _, tt := range cleanerTests {
+		t.Run(tt.in, func(t *testing.T) {
+			buf := bytes.Buffer{}
+			buf.WriteString(tt.in)
+			cmdLines, _, _ := matchRegex(optcmd, &buf)
+
+			if (!reflect.DeepEqual(cmdLines, tt.out)) {
+				t.Errorf("Commands not equal!\nGot:  %v\nWant: %v", cmdLines, tt.out)
+			}
+		})
+	}
+
+}

